app/obolapi/api.go

```diff
@@ -105,27 +105,58 @@ func launchpadURLPath(lock cluster.Lock) string {
 	return fmt.Sprintf(launchpadReturnPathFmt, lock.LockHash)
 }
 
-func httpPost(ctx context.Context, url *url.URL, b []byte) error {
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url.String(), bytes.NewReader(b))
+func httpPost(ctx context.Context, url *url.URL, body []byte, headers map[string]string) error {
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url.String(), bytes.NewReader(body))
 	if err != nil {
 		return errors.Wrap(err, "new POST request with ctx")
 	}
 	req.Header.Add("Content-Type", "application/json")
+	for key, val := range headers {
+		req.Header.Set(key, val)
+	}
 
 	res, err := new(http.Client).Do(req)
 	if err != nil {
-		return errors.Wrap(err, "failed to call POST endpoint")
+		return errors.Wrap(err, "call POST endpoint")
 	}
 	defer res.Body.Close()
 
-	data, err := io.ReadAll(res.Body)
+	if res.StatusCode/100 != 2 {
+		data, err := io.ReadAll(res.Body)
+		if err != nil {
+			return errors.Wrap(err, "read POST response")
```

Comment:
It would be great to catch the status code as well as headers when an error occurs, or maybe log them as debug. Also, verbose logging may be printing the body as base64 or json (depending on content-type). I remember I needed this a few times and patched this code locally to do so.

Reply:
> It would be great to catch the status code as well as headers when an error occurs, or maybe log them as debug.
Indeed! Added it.

> Also, verbose logging may be printing the body as base64 or json (depending on content-type).
What do you mean here?

Reply:
I realized we don't have verbose logging level, debug is the minimum.
I meant the ability to intercept HTTP requests and responses (bodies) when we need to intercept traffic between stack components.. in addition to HTTP status code, URL and Headers. To see the full picture.

Reply:
Ah, I see now. Yes having some sort of a "trace" logging would help in a lot of scenarios I believe. We can probably add it on the roadmap to add it throughout the project?

Reply:
Maybe if your Wireshark experience was successful then we don't need to add anything here. Anyway this out of scope for the current work.